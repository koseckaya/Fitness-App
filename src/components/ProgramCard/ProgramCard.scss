@import '../../styles/vars';

.program-card {
  box-sizing: border-box;
  display: flex;
  margin-bottom: 80px;
  &__title {
    font-size: 18px;
    font-weight: 500;
  }
  &__inner{
    margin-bottom: 16px;
    position: relative;
    transition: all 0.5s;
    color: $gray;
     overflow: hidden;
     cursor: pointer;
    &:hover {
      .card__content {
        bottom: 15px;
      }
      .card__image {
         filter: brightness(0.6);
      }
    }
  }
}
.card {
  &__image {
    height: 490px;
    width: 400px;
    border-radius: 15px;
     transition: all 0.5s;
     object-fit: cover;
  }
  &__content {
    position: absolute;
    display: flex;
    flex-direction: column;
    bottom: -44px;
    left: 15px;
    gap: 10px;
    transition: all 0.5s;
  }
  &__duration, &__minutes{
    text-transform: lowercase;
    background: #fff;
   border-radius: 15px;
    padding: 5px;
    color: $color-primary;
  }
  &__duration {
    width: 57px;
  }
   &__minutes {
    width: 106px;
  }
}