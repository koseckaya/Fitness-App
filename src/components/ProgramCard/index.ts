export { default as ProgramCard } from './ProgramCard';
export type { Props as ProgramCardProps } from './ProgramCard';
