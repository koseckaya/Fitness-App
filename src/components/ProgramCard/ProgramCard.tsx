import { FC } from 'react';
import './ProgramCard.scss';

export type Props = {
    className?: string;
    title?: string;
    image?: string;
    url?: string;
    duration?: string;
    time?: string;
    type?: string[];
    equipment?: string[];
};

const ProgramCard: FC<Props> = ({ title, image,  url, duration, time, type, equipment }: Props) => (
    <div className='program-card'>
        <a href={url}>
            <div className='program-card__inner'>
                <img src={image} alt={title} className='card__image' />
                <div className='card__content'>
                    {duration ?  <span className='card__duration'>{duration}</span> : null }
                    {time ?  <span className='card__minutes'>{time}</span> : null }
                    {type ? <div className='card__type'>Type: { type?.join(', ') }</div> : null }
                    {equipment ? <div className='card__type'>Equipment: { equipment?.join(', ') }</div>  : null }
                </div>
            </div>
            <span className='program-card__title'>{title}</span>
        </a>
    </div>
);

export default ProgramCard;
