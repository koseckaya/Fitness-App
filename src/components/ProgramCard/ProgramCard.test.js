import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import ProgramCard from './ProgramCard';

const props = {
    title: "Weight Loss Challenge",
    image: "https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/971c80f0-87c8-11ed-8684-75e08be900e2.jpeg",
    url: "https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/97331630-87c8-11ed-8684-75e08be900e2.jpeg",
    days: 28,
    time: [20, 50],
    type: ["Weight Loss", "Full Body"],
    equip: ["Fitness Mat", "Dumbbells", "Resistance Bands"],
};

describe("ProgramCard component", () => {
    it('should render component with default props', () => {
        const component = renderer
            .create(<Router><ProgramCard {...props}></ProgramCard></Router>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });

    it('should render component without prop image', () => {
        const newProps = { ...props, image: '' }
        const component = renderer
            .create(<Router><ProgramCard {...newProps}></ProgramCard></Router>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without prop title', () => {
        const newProps = { ...props, title: '' }
        const component = renderer
            .create(<Router><ProgramCard {...newProps}></ProgramCard></Router>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without prop url', () => {
        const newProps = { ...props, url: '' }
        const component = renderer
            .create(<Router><ProgramCard {...newProps}></ProgramCard></Router>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without prop days', () => {
        const newProps = { ...props, days: 0 }
        const component = renderer
            .create(<Router><ProgramCard {...newProps}></ProgramCard></Router>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without prop time', () => {
        const newProps = { ...props, time: [] }
        const component = renderer
            .create(<Router><ProgramCard {...newProps}></ProgramCard></Router>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without prop equip', () => {
        const newProps = { ...props, equip: [] }
        const component = renderer
            .create(<Router><ProgramCard {...newProps}></ProgramCard></Router>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
    it('should render component without prop type', () => {
        const newProps = { ...props, type: [] }
        const component = renderer
            .create(<Router><ProgramCard {...newProps}></ProgramCard></Router>)
            .toJSON();
        expect(component).toMatchSnapshot();
    });
})