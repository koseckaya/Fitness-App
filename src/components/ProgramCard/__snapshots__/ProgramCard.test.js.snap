// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgramCard component should render component with default props 1`] = `
<div
  className="program-card"
>
  <a
    href="/programshttps://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/97331630-87c8-11ed-8684-75e08be900e2.jpeg"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        alt="Weight Loss Challenge"
        className="card__image"
        src="https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/971c80f0-87c8-11ed-8684-75e08be900e2.jpeg"
      />
      <div
        className="card__content"
      >
        <span
          className="card__duration"
        >
          28
           days
        </span>
        <span
          className="card__minutes"
        >
          20
          -
          50
           min/day
        </span>
        <div
          className="card__type"
        >
          Type: 
          Weight Loss, Full Body
        </div>
        <div
          className="card__type"
        >
          Equipment: 
          Fitness Mat, Dumbbells, Resistance Bands
        </div>
      </div>
    </div>
    <span
      className="program-card__title"
    >
      Weight Loss Challenge
    </span>
  </a>
</div>
`;

exports[`ProgramCard component should render component without prop days 1`] = `
<div
  className="program-card"
>
  <a
    href="/programshttps://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/97331630-87c8-11ed-8684-75e08be900e2.jpeg"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        alt="Weight Loss Challenge"
        className="card__image"
        src="https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/971c80f0-87c8-11ed-8684-75e08be900e2.jpeg"
      />
      <div
        className="card__content"
      >
        <span
          className="card__minutes"
        >
          20
          -
          50
           min/day
        </span>
        <div
          className="card__type"
        >
          Type: 
          Weight Loss, Full Body
        </div>
        <div
          className="card__type"
        >
          Equipment: 
          Fitness Mat, Dumbbells, Resistance Bands
        </div>
      </div>
    </div>
    <span
      className="program-card__title"
    >
      Weight Loss Challenge
    </span>
  </a>
</div>
`;

exports[`ProgramCard component should render component without prop equip 1`] = `
<div
  className="program-card"
>
  <a
    href="/programshttps://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/97331630-87c8-11ed-8684-75e08be900e2.jpeg"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        alt="Weight Loss Challenge"
        className="card__image"
        src="https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/971c80f0-87c8-11ed-8684-75e08be900e2.jpeg"
      />
      <div
        className="card__content"
      >
        <span
          className="card__duration"
        >
          28
           days
        </span>
        <span
          className="card__minutes"
        >
          20
          -
          50
           min/day
        </span>
        <div
          className="card__type"
        >
          Type: 
          Weight Loss, Full Body
        </div>
        <div
          className="card__type"
        >
          Equipment: 
        </div>
      </div>
    </div>
    <span
      className="program-card__title"
    >
      Weight Loss Challenge
    </span>
  </a>
</div>
`;

exports[`ProgramCard component should render component without prop image 1`] = `
<div
  className="program-card"
>
  <a
    href="/programshttps://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/97331630-87c8-11ed-8684-75e08be900e2.jpeg"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        alt="Weight Loss Challenge"
        className="card__image"
        src=""
      />
      <div
        className="card__content"
      >
        <span
          className="card__duration"
        >
          28
           days
        </span>
        <span
          className="card__minutes"
        >
          20
          -
          50
           min/day
        </span>
        <div
          className="card__type"
        >
          Type: 
          Weight Loss, Full Body
        </div>
        <div
          className="card__type"
        >
          Equipment: 
          Fitness Mat, Dumbbells, Resistance Bands
        </div>
      </div>
    </div>
    <span
      className="program-card__title"
    >
      Weight Loss Challenge
    </span>
  </a>
</div>
`;

exports[`ProgramCard component should render component without prop time 1`] = `
<div
  className="program-card"
>
  <a
    href="/programshttps://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/97331630-87c8-11ed-8684-75e08be900e2.jpeg"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        alt="Weight Loss Challenge"
        className="card__image"
        src="https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/971c80f0-87c8-11ed-8684-75e08be900e2.jpeg"
      />
      <div
        className="card__content"
      >
        <span
          className="card__duration"
        >
          28
           days
        </span>
        <span
          className="card__minutes"
        >
          -
           min/day
        </span>
        <div
          className="card__type"
        >
          Type: 
          Weight Loss, Full Body
        </div>
        <div
          className="card__type"
        >
          Equipment: 
          Fitness Mat, Dumbbells, Resistance Bands
        </div>
      </div>
    </div>
    <span
      className="program-card__title"
    >
      Weight Loss Challenge
    </span>
  </a>
</div>
`;

exports[`ProgramCard component should render component without prop title 1`] = `
<div
  className="program-card"
>
  <a
    href="/programshttps://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/97331630-87c8-11ed-8684-75e08be900e2.jpeg"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        alt=""
        className="card__image"
        src="https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/971c80f0-87c8-11ed-8684-75e08be900e2.jpeg"
      />
      <div
        className="card__content"
      >
        <span
          className="card__duration"
        >
          28
           days
        </span>
        <span
          className="card__minutes"
        >
          20
          -
          50
           min/day
        </span>
        <div
          className="card__type"
        >
          Type: 
          Weight Loss, Full Body
        </div>
        <div
          className="card__type"
        >
          Equipment: 
          Fitness Mat, Dumbbells, Resistance Bands
        </div>
      </div>
    </div>
    <span
      className="program-card__title"
    />
  </a>
</div>
`;

exports[`ProgramCard component should render component without prop type 1`] = `
<div
  className="program-card"
>
  <a
    href="/programshttps://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/97331630-87c8-11ed-8684-75e08be900e2.jpeg"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        alt="Weight Loss Challenge"
        className="card__image"
        src="https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/971c80f0-87c8-11ed-8684-75e08be900e2.jpeg"
      />
      <div
        className="card__content"
      >
        <span
          className="card__duration"
        >
          28
           days
        </span>
        <span
          className="card__minutes"
        >
          20
          -
          50
           min/day
        </span>
        <div
          className="card__type"
        >
          Type: 
        </div>
        <div
          className="card__type"
        >
          Equipment: 
          Fitness Mat, Dumbbells, Resistance Bands
        </div>
      </div>
    </div>
    <span
      className="program-card__title"
    >
      Weight Loss Challenge
    </span>
  </a>
</div>
`;

exports[`ProgramCard component should render component without prop url 1`] = `
<div
  className="program-card"
>
  <a
    href="/programs"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        alt="Weight Loss Challenge"
        className="card__image"
        src="https://static.chloeting.com/programs/63ae15573e8cb6226b0c40c0/971c80f0-87c8-11ed-8684-75e08be900e2.jpeg"
      />
      <div
        className="card__content"
      >
        <span
          className="card__duration"
        >
          28
           days
        </span>
        <span
          className="card__minutes"
        >
          20
          -
          50
           min/day
        </span>
        <div
          className="card__type"
        >
          Type: 
          Weight Loss, Full Body
        </div>
        <div
          className="card__type"
        >
          Equipment: 
          Fitness Mat, Dumbbells, Resistance Bands
        </div>
      </div>
    </div>
    <span
      className="program-card__title"
    >
      Weight Loss Challenge
    </span>
  </a>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="program-card"
>
  <a
    href="/programsundefined"
    onClick={[Function]}
  >
    <div
      className="program-card__inner"
    >
      <img
        className="card__image"
      />
      <div
        className="card__content"
      />
    </div>
    <span
      className="program-card__title"
    />
  </a>
</div>
`;
